### DISCUSSION TITLE
Problem during upgrading to Angular 19

### DISCUSSION THREAD

**Discussion by balagex** [primefaces/community#3148] [PrimeNG]
Hi,

I have a small project where I used Angular 17 with Primeng 17. I tried to upgrade angular version to 19 in two steps.

The angular 17 to 18 went fine, but when I started to upgrade to 19 (ng update @angular/core@19 @angular/cli@19 @angular/cdk@19), I got a primeng related error:

" Package "primeng" has an incompatible peer dependency to "zone.js" (requires "~0.14.0", would install "0.15.0")"

What should I do?

The actual dependecies in package.jon:
  "dependencies": {
    "@angular/animations": "^18.2.13",
    "@angular/cdk": "^18.2.14",
    "@angular/common": "^18.2.13",
    "@angular/compiler": "^18.2.13",
    "@angular/core": "^18.2.13",
    "@angular/forms": "^18.2.13",
    "@angular/platform-browser": "^18.2.13",
    "@angular/platform-browser-dynamic": "^18.2.13",
    "@angular/router": "^18.2.13",
    "@ngx-translate/core": "^15.0.0",
    "@ngx-translate/http-loader": "^8.0.0",
    "firebase": "^10.13.2",
    "lodash": "^4.17.21",
    "primeicons": "^6.0.1",
    "primeng": "^17.18.12",
    "rxjs": "~7.8.1",
    "tslib": "^2.7.0",
    "zone.js": "~0.14.10"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^18.2.12",
    "@angular/cli": "^18.2.12",
    "@angular/compiler-cli": "^18.2.13",
    "@angular/language-service": "18.2.13",
    "@types/jasmine": "~5.1.4",
    "@types/lodash": "^4.17.10",
    "jasmine-core": "~5.1.2",
    "karma": "~6.4.0",
    "karma-chrome-launcher": "~3.2.0",
    "karma-coverage": "~2.2.1",
    "karma-jasmine": "~5.1.0",
    "karma-jasmine-html-reporter": "~2.1.0",
    "typescript": "~5.5.4"
  }

Can I use primeng 17 with Angular 19 or need to upgrade to primeng 18 first?

As I see, there is no "stable" primeng 18 version, only rc or beta.  Using primeng 18 rc can resolve this zone.js version problem?

**Comment by mertsincan** [comment]
Please see;
https://github.com/primefaces/primeng/issues/16827
